contrib/src/test/java/com/datatorrent/contrib/parser/StreamingJsonParserTest.java

```diff
@@ -0,0 +1,269 @@
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.parser;
+
+import java.util.List;
+import java.util.ListIterator;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.python.google.common.collect.Lists;
+
+import com.datatorrent.api.Attribute;
+import com.datatorrent.api.Context;
+import com.datatorrent.lib.helper.TestPortContext;
+import com.datatorrent.lib.testbench.CollectorTestSink;
+
+public class StreamingJsonParserTest
+{
+  public static final String fieldInfoInitMap = "id:id:INTEGER," + "name:name:STRING," + "gpa:gpa:DOUBLE";
+
+  public static final String nestedFieldInfoMap = "id:id:INTEGER," + "name:name:STRING," + "gpa:gpa:DOUBLE,"
+      + "streetAddress:streetAddress:STRING," + "city:city:STRING," + "state:state:STRING,"
+      + "postalCode:postalCode:STRING";
+
+  public static final String invalidFieldInfoMap = "Field1:id:INTEGER," + "name:name:STRING," + "gpa:gpa:DOUBLE";
+
+  CollectorTestSink<Object> outputSink = new CollectorTestSink<Object>();
+  CollectorTestSink<Object> errorSink = new CollectorTestSink<Object>();
+
+  StreamingJsonParser jsonParser = new StreamingJsonParser();
+
+  private List<String> recordList = null;
+
+  public class TestMeta extends TestWatcher
+  {
+    Context.OperatorContext context;
+    Context.PortContext portContext;
+
+    @Override
+    protected void starting(org.junit.runner.Description description)
+    {
+      Attribute.AttributeMap portAttributes = new Attribute.AttributeMap.DefaultAttributeMap();
+      portAttributes.put(Context.PortContext.TUPLE_CLASS, Person.class);
+      portContext = new TestPortContext(portAttributes);
+      super.starting(description);
+      jsonParser.output.setSink(outputSink);
+      jsonParser.err.setSink(errorSink);
+    }
+
+    @Override
+    protected void finished(Description description)
+    {
+      jsonParser.teardown();
+    }
+  }
+
+  @Rule
+  public TestMeta testMeta = new TestMeta();
+
+  @Test
+  public void testReads() throws Exception
+  {
+    int count = 5;
+    createReaderInput(count);
+    jsonParser.setPojoClass(Person.class);
```

Comment:
We don't need to call this method once the initialization is done in setup method of output port